core/src/main/java/org/elasticsearch/search/aggregations/AggregatorBase.java

```diff
@@ -96,6 +102,14 @@ public boolean needsScores() {
                 return false; // unreachable
             }
         };
+        try {
+            this.breakerService
+                    .getBreaker(CircuitBreaker.BUCKET)
+                    .addEstimateBytesAndMaybeBreak(1024, "<agg [" + name + "]>");
+        } catch (CircuitBreakingException cbe) {
+            this.failed = true;
+            throw cbe;
+        }
```

Comment:
It is also fine to use this 1024 constant, even if overestimated. Aggregations that hurt us most are those that create many aggregator objects so it sends the right incentive.